qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,252 @@
+
+from qiskit_experiments.exceptions import QiskitError
+from qiskit_experiments.test.utils import FakeJob
+
+
+class IQPulseBackend(BackendV2):
+    """Pulse Simulator abstract class"""
+
+    def __init__(
+        self,
+        static_hamiltonian: np.ndarray,
+        hamiltonian_operators: np.ndarray,
+        dt: Optional[float] = 0.1 * 1e-9,
+        **kwargs,
+    ):
+        """Hamiltonian and operators is the Qiskit Dynamics object"""
+        super().__init__(
+            None,
+            name="PulseBackendV2",
+            description="A PulseBackend simulator",
+            online_date=datetime.datetime.utcnow(),
+            backend_version="0.0.1",
+        )
+        self._defaults = PulseDefaults.from_dict(
+            {
+                "qubit_freq_est": [0],
+                "meas_freq_est": [0],
+                "buffer": 0,
+                "pulse_library": [],
+                "cmd_def": [],
+            }
+        )
+
+        self.converter = None
+
+        self.static_hamiltonian = static_hamiltonian
+        self.hamiltonian_operators = hamiltonian_operators
+        self.solver = Solver(self.static_hamiltonian, self.hamiltonian_operators, **kwargs)
+        self._target = Target(dt=dt, granularity=16)
+        self.gound_state = np.zeros(self.solver.model.dim)
+        self.gound_state[0] = 1
+        self.y_0 = np.eye(self.solver.model.dim)
+
+    @property
+    def default_pulse_unitaries(self) -> Dict[Tuple, np.array]:
+        """Return the default unitary matrices of the backend."""
+        return self._simulated_pulse_unitaries
+
+    @default_pulse_unitaries.setter
+    def default_pulse_unitaries(self, unitaries: Dict[Tuple, np.array]):
+        """Set the default unitary pulses this allows the tests to simulate the pulses only once."""
+        self._simulated_pulse_unitaries = unitaries
+
+    @property
+    def target(self):
+        return self._target
+
+    @property
+    def max_circuits(self):
+        return None
+
+    def defaults(self):
+        """return backend defaults"""
+        return self._defaults
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=1024)
+
+    @staticmethod
+    def _get_info(instruction: CircuitInstruction) -> Tuple[Tuple(int), Tuple(float), str]:
+        p_dict = instruction.operation
+        qubit = tuple(int(str(val)[-2]) for val in instruction.qubits)
+        params = tuple(float(val) for val in p_dict.params)
+        return qubit, params, p_dict.name
+
+    @staticmethod
+    def _state_vector_to_result(
+        state: Union[Statevector, np.ndarray],
+        shots: Optional[int] = 1024,
+        meas_return: Optional[MeasReturnType] = 0,
+        meas_level: Optional[MeasLevel] = 0,
+    ) -> Union[Dict[str, int], complex]:
+        """Convert the state vector to IQ data or counts."""
+
+        if meas_level == MeasLevel.CLASSIFIED:
+            measurement = Statevector(state).sample_counts(shots)
```

Comment:
Statevector take np.array or list or Statevector or Operator or QuantumCircuit. However, we have to give a matrix form to the state which Statevector cannot take.